config/src/test/java/org/springframework/security/SpringSecurityCoreVersionSerializableTests.java

```diff
@@ -309,6 +311,7 @@ class SpringSecurityCoreVersionSerializableTests {
 				(r) -> TestSaml2PostAuthenticationRequests.create());
 		generatorByClassName.put(Saml2RedirectAuthenticationRequest.class,
 				(r) -> TestSaml2RedirectAuthenticationRequests.create());
+		generatorByClassName.put(RelyingPartyRegistration.class, (r) -> TestRelyingPartyRegistrations.full().build());
```

Comment:
Please run the `serializeCurrentVersionClasses` test so that the appropriate `.serialized` file gets created. You can see #16276 for more details.